hack/generator/pkg/codegen/pipeline_create_arm_types.go

```diff
@@ -151,7 +159,14 @@ func iterDefs(
 				return nil, err
 			}
 
-			newDefs.Add(newDef)
+			if existingNewDef, ok := newDefs[newDef.Name()]; ok {
+				if !existingNewDef.Type().Equals(newDef.Type()) {
+					return nil, errors.Errorf("mismatch in generated ARM type %v (existing: %v, new: %v)", newDef.Name(), existingNewDef.Type(), newDef.Type())
```

Comment:
Doesn't `Types.Add` panic if there's a collision?

Is there a reason we want to bypass that for a specific and handleable error here?

Reply:
IIRC correctly, in this case the type can be reachable by multiple paths, so we end up creating the same type multiple times.

Reply:
Do we want a comment to that effect?

Reply:
whatever was causing this has been fixed, so I'm removing the additional code